src/compiler/sys/typescript/typescript-sys.ts

```diff
@@ -55,8 +55,13 @@ export const patchTsSystemFileSystem = (
   };
 
   tsSys.directoryExists = (p) => {
-    const s = inMemoryFs.statSync(p);
-    return s.isDirectory;
+    if (inMemoryFs) {
```

Comment:
I'm not entirely sure I understand why we need to fallback to `compilerSys` for this PR based on the existing comments/PR summary. Can you explain that/add it to either please?

Reply:
yes, sorry should have probably left on draft while I was sorting just that out

Reply:
The reason basically has to do with these calls to `patchTypescript`, here:

https://github.com/ionic-team/stencil/blob/f0781d29160d794fcd1315c2631a286437d71fb7/src/compiler/transpile.ts#L43-L43

and here:

https://github.com/ionic-team/stencil/blob/f0781d29160d794fcd1315c2631a286437d71fb7/src/compiler/transpile.ts#L75

Because we pass in `null` here we actually can't count on the `inMemoryFs` argument to `patchTsFileSystem` being defined in here :(

I was considering addressing STENCIL-728 as part of this PR but thought it better to save that for another, however I did have to make these edits in `typescript-sys` because these methods are called by TypeScript when transpiling a spec file.

Reply:
I added some comments to this file (as well as TODOs for STENCIL-728) just noting why we need to make the check